src/elements/List/List.js

```diff
@@ -23,124 +23,144 @@ import ListList from './ListList'
+    floated: PropTypes.oneOf(SUI.FLOATS),
 
-  /** Additional classes. */
-  className: PropTypes.string,
+    /** A list can be formatted to have items appear horizontally. */
+    horizontal: PropTypes.bool,
 
-  /** A list can show divisions between content. */
-  divided: PropTypes.bool,
+    /** A list can be inverted to appear on a dark background. */
+    inverted: PropTypes.bool,
 
-  /** An list can be floated left or right. */
-  floated: PropTypes.oneOf(SUI.FLOATS),
+    /** Shorthand array of props for ListItem. */
+    items: customPropTypes.collectionShorthand,
 
-  /** A list can be formatted to have items appear horizontally. */
-  horizontal: PropTypes.bool,
+    /** A list can be specially formatted for navigation links. */
+    link: PropTypes.bool,
 
-  /** A list can be inverted to appear on a dark background. */
-  inverted: PropTypes.bool,
+    /**
+     * onClick handler for ListItem. Mutually exclusive with children.
+     *
+     * @param {SyntheticEvent} event - React's original SyntheticEvent.
+     * @param {object} data - All item props.
+     */
+    onItemClick: customPropTypes.every([
+      customPropTypes.disallow(['children']),
+      PropTypes.func,
+    ]),
 
-  /** Shorthand array of props for ListItem. */
-  items: customPropTypes.collectionShorthand,
+    /** A list can be ordered numerically. */
+    ordered: PropTypes.bool,
 
-  /** A list can be specially formatted for navigation links. */
-  link: PropTypes.bool,
+    /** A list can relax its padding to provide more negative space. */
+    relaxed: PropTypes.oneOfType([
+      PropTypes.bool,
+      PropTypes.oneOf(['very']),
+    ]),
 
-  /** A list can be ordered numerically. */
-  ordered: PropTypes.bool,
+    /** A selection list formats list items as possible choices. */
+    selection: PropTypes.bool,
 
-  /** A list can relax its padding to provide more negative space. */
-  relaxed: PropTypes.oneOfType([
-    PropTypes.bool,
-    PropTypes.oneOf(['very']),
-  ]),
+    /** A list can vary in size. */
+    size: PropTypes.oneOf(SUI.SIZES),
 
-  /** A selection list formats list items as possible choices. */
-  selection: PropTypes.bool,
+    /** An element inside a list can be vertically aligned. */
+    verticalAlign: PropTypes.oneOf(SUI.VERTICAL_ALIGNMENTS),
+  }
 
-  /** A list can vary in size. */
-  size: PropTypes.oneOf(SUI.SIZES),
+  static _meta = {
+    name: 'List',
+    type: META.TYPES.ELEMENT,
+  }
 
-  /** An element inside a list can be vertically aligned. */
-  verticalAlign: PropTypes.oneOf(SUI.VERTICAL_ALIGNMENTS),
+  static Content = ListContent
+  static Description = ListDescription
+  static Header = ListHeader
+  static Icon = ListIcon
+  static Item = ListItem
+  static List = ListList
+
+  handleItemOverrides = predefinedProps => ({
+    onClick: (e, itemProps) => {
+      _.invoke(predefinedProps, 'onClick', e, itemProps)
+      _.invoke(this.props, 'onItemClick', e, itemProps)
+    },
+  })
```

Comment:
I've updated handlers there like in #1428. I also removed `index` there, I think we don't really need it there.